libnmstate/nm/checkpoint.py

```diff
@@ -43,6 +46,12 @@ class NMCheckPointCreationError(NMCheckPointError):
     pass
 
 
+class NMCheckPointPermissionError(NMCheckPointError):
```

Comment:
Isn't this a specific creation error? If so, it needs to be a child of `NMCheckPointCreationError`.

Reply:
Currently it is only used in the creation step but commit and rollback should probably handle this the same way and I do not think we need a separate exception for each case.